dags/hivemind_etl_helpers/ingestion_pipeline.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+from dotenv import load_dotenv
+from llama_index.core import MockEmbedding
+from llama_index.core.ingestion import (DocstoreStrategy, IngestionCache,
+                                        IngestionPipeline)
+from llama_index.core.node_parser import SemanticSplitterNodeParser
+from llama_index.storage.docstore.postgres import PostgresDocumentStore
+from llama_index.storage.kvstore.redis import RedisKVStore as RedisCache
+from llama_index.vector_stores.postgres import PGVectorStore
+from tc_hivemind_backend.db.credentials import load_postgres_credentials
+from tc_hivemind_backend.embeddings.cohere import CohereEmbedding
+
+from dags.hivemind_etl_helpers.src.db.gdrive.db_utils import setup_db
+
+
+class CustomIngestionPipeline:
+    def __init__(self, community_id: str, table_name: str,
+                 testing: bool = False):
+        load_dotenv()
+        postgres_credentials = load_postgres_credentials()
+        self.redis_host = os.getenv("REDIS_HOST")
+        self.redis_port = os.getenv("REDIS_PORT")
+
+        self.table_name = table_name
+        self.dbname = f"community_{community_id}"
+        self.db_config = {
+            "host": postgres_credentials["host"],
+            "password": postgres_credentials["password"],
+            "port": postgres_credentials["port"],
+            "user": postgres_credentials["user"],
+        }
+        self.community_id = community_id
+        self.embed_model = CohereEmbedding()
+        if testing:
+            self.embed_model = MockEmbedding(embed_dim=1024)
+
+    def run_pipeline(self, docs):
+        setup_db(community_id=self.community_id)
+        pipeline = IngestionPipeline(
+            transformations=[
+                SemanticSplitterNodeParser(embed_model=self.embed_model),
+                self.embed_model,
+            ],
+            docstore=PostgresDocumentStore.from_params(
+                **self.db_config,
+                database=self.dbname,
+                table_name=self.table_name + "_docstore",
+            ),
+            vector_store=PGVectorStore.from_params(
+                **self.db_config,
+                database=self.dbname,
+                table_name=self.table_name,
+                embed_dim=1024,
+            ),
+            cache=IngestionCache(
+                cache=RedisCache.from_host_and_port(self.redis_host,
+                                                    self.redis_port),
+                collection=self.dbname + "_ingestion_cache",
```

Comment:
Please consider updating the name to `self.dbname + f"_{self.table_name}" + "_ingestion_cache"`.

Reply:
The reason is to avoid conflict between multiple communities' data.